src/controllers/parse.controller.ts

```diff
@@ -0,0 +1,36 @@
+import { Request, Response, Router } from 'express';
+import { AsyncAPIDocument } from '@asyncapi/parser';
+
+import { validationMiddleware } from '../middlewares/validation.middleware';
+
+import { Controller} from '../interfaces';
+
+/**
+ * Controller which exposes the Parser functionality, to parse the AsyncAPI document.
+ */
+export class ParseController implements Controller {
+  public basepath = '/parse';
+
+  private async parse(req: Request, res: Response) {
+    const stringified = AsyncAPIDocument.stringify(req.asyncapi?.parsedDocument);
+    res.status(200).json({
+      parsed: stringified,
+    });
```

Comment:
Should we add an `if` statement in case `req.asyncapi` is null or undefined? Because in that case calling `stringify` will result in an error here right https://github.com/asyncapi/parser-js/blob/dcee64944343f6fb74566e3905680d133946332f/lib/models/asyncapi.js#L247 ?
And if no asyncapi object is inside `req` then maybe we should return a `400` or `422`, @magicmatatjahu wdyt?

Reply:
First, the `validationMiddleware` validates the AsyncAPI document for us. If it is invalid then the controller will not run at all, so we have 100% sure that `req.asyncapi?.parsedDocument` is not null/undefined, but due to fact that we have custom typings we have to treat `req.asyncapi` as optional field.